modules/EPMC.py

```diff
@@ -0,0 +1,183 @@
+            (col('type') == "GP-DS") & 
+            (col('isMapped') == True) & 
+            (col('pmid').isNotNull()) & 
+            (length(col('text')) < 600) &
+            (col('label1').isin(filtered_target_terms) == False)
+        )
+        # Renaming columns:
+        .withColumnRenamed("keywordId1", "targetFromSourceId")
+        .withColumnRenamed("keywordId2", "diseaseFromSourceMappedId")
+        .withColumnRenamed("label1", "targetFromSource")
+        .withColumnRenamed("label2", "diseaseFromSource")
+
+        .withColumn('tmp',col('pmid').cast(StringType()))
+    )
+
+    # Report on the number of diseases, targets and associations:
+    logging.info(f'Number of publications: {filtered_cooccurrence_df.select(col("tmp")).distinct().count()}')
+    logging.info(f'Number of targets: {filtered_cooccurrence_df.select(col("targetFromSourceId")).distinct().count()}')
+    logging.info(f'Number of diseases: {filtered_cooccurrence_df.select(col("diseaseFromSourceMappedId")).distinct().count()}')
+    logging.info(f'Number of associations: {filtered_cooccurrence_df.select(col("diseaseFromSourceMappedId"), col("targetFromSourceId")).dropDuplicates().count()}')
+
+    # Aggregating cooccurrence, get score apply filter:    
+    aggregated_df = (
+        filtered_cooccurrence_df
+
+        # Aggregating data by publication, target and disease:
+        .groupBy(['pmid', 'targetFromSourceId', 'diseaseFromSourceMappedId'])
+        .agg(
+            first(col("targetFromSource")).alias("targetFromSource"),
+            first(col("diseaseFromSource")).alias("diseaseFromSource"),
+            collect_set(col('tmp')).alias('literature'),
+            collect_set(
+                struct(  
+                    col("text"),
+                    col('start1').alias('tStart'),
+                    col("end1").alias('tEnd'),
+                    col('start2').alias('dStart'),
+                    col("end2").alias('dEnd'),
+                    col('section')
+                )
+            ).alias('textMiningSentences'),
+            sum(col('evidence_score')).alias('resourceScore')
+        )
+
+        # Summarizing all scores then filter for evidence string with at least score == 2
+        # .withColumn('resourceScore', sum(col('evidence_score')).over(w))
+        # .withColumn('resourceScores', collect_list(col('evidence_score')).over(w))
+        .filter(col('resourceScore') > 1) 
+    )
+
+    # Report number of evidence:
+    logging.info(f'Number of evidence: {aggregated_df.count()}')
+
+    # Final formatting and saving data:
+    (
+        aggregated_df
+
+        # Adding literal columns:
+        .withColumn('datasourceId',lit('europepmc'))
+        .withColumn('datatypeId',lit('literature'))
+
+        # Reorder columns:
+        .select(["datasourceId", "datatypeId", "targetFromSource", "targetFromSourceId", 'resourceScore',
+                "diseaseFromSource","diseaseFromSourceMappedId","literature","textMiningSentences"])
```

Comment:
Final place with the literal mismatch :)

```suggestion
        # Reorder columns:
        .select(['datasourceId', 'datatypeId', 'targetFromSource', 'targetFromSourceId', 'resourceScore',
                 'diseaseFromSource', 'diseaseFromSourceMappedId', 'literature', 'textMiningSentences'])
```